src/Microsoft.AspNetCore.Mvc.Core/Internal/ExcludeBindingMetadataProvider.cs

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.AspNetCore.Mvc.ModelBinding.Metadata;
+
+namespace Microsoft.AspNetCore.Mvc.Internal
+{
+    /// <summary>
+    /// An <see cref="IBindingMetadataProvider"/> which configures <see cref="ModelMetadata.IsBindingAllowed"/> to
+    /// <c>false</c> for matching types.
+    /// </summary>
+    public class ExcludeBindingMetadataProvider : IBindingMetadataProvider
+    {
+        private readonly Type _type;
+
+        /// <summary>
+        /// Creates a new <see cref="ExcludeBindingMetadataProvider"/> for the given <paramref name="type"/>.
+        /// </summary>
+        /// <param name="type">
+        /// The <see cref="Type"/>. All properties of this <see cref="Type"/> will have
+        /// <see cref="ModelMetadata.IsBindingAllowed"/> set to <c>false</c>.
+        /// </param>
+        public ExcludeBindingMetadataProvider(Type type)
+        {
+            if (type == null)
+            {
+                throw new ArgumentNullException(nameof(type));
+            }
+
+            _type = type;
+        }
+
+        /// <inheritdoc />
+        public void CreateBindingMetadata(BindingMetadataProviderContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            // No-op if the metadata is not for the target type
+            if (!_type.IsAssignableFrom(context.Key.ModelType))
+            {
+                return;
+            }
+
+            if (context.Key.MetadataKind == ModelMetadataKind.Property)
+            {
+                // BindingBehavior can fall back to attributes on the Container Type, but we should ignore
+                // attributes on the Property Type.
+                var bindingBehavior = context.PropertyAttributes.OfType<BindingBehaviorAttribute>().FirstOrDefault();
+                if (bindingBehavior == null)
+                {
+                    bindingBehavior =
+                        context.Key.ContainerType.GetTypeInfo()
+                        .GetCustomAttributes(typeof(BindingBehaviorAttribute), inherit: true)
+                        .OfType<BindingBehaviorAttribute>()
+                        .FirstOrDefault();
+                }
```

Comment:
I don't get why any of this is needed. Ordering can solve this problem if the user needs to override this behavior. I'd rather not have scary reflection that doesn't follow the typical pattern.